examples/sampling/graphbolt/link_prediction.py

```diff
@@ -340,6 +340,8 @@ def train(args, model, graph, features, train_set):
 
             total_loss += loss.item()
             if step + 1 == args.early_stop:
+                # Early stopping requires a new dataloader to reset its state.
+                dataloader = create_dataloader(args, graph, features, train_set)
```

Comment:
Will this cause retrain on the same nodes?

Reply:
I think it should shuffle again because it is the training dataloader. I expect the behaviour to be the same as using the same dataloader again. @Rhett-Ying is this true?

Reply:
Ah I see.

Reply:
It doesn't. The link in the PR description has more details about the same issue someone else experienced

Reply:
And a potential fix to reset states without recreating the dataloader.

Reply:
This issue does not arise when we exhaust all items in the dataloader because in that case, the buffers get flushed.